### ISSUE TITLE
IOException: Unexpected end of stream

### ISSUE THREAD

**Issue by cliftonm** [Microsoft/Docker.DotNet#204]
Very simple test program:

           static void Main(string[] args)
		{
			DockerClient client = new DockerClientConfiguration(new Uri("tcp://192.168.99.100:2376")).CreateClient();
			Get(client);
			Console.ReadLine();
		}

		static async void Get(DockerClient client)
		{
			IList<ContainerListResponse> containers = await client.Containers.ListContainersAsync(
				new ContainersListParameters()
				{
					Limit = 10,
				});

			Console.WriteLine("Foobar");
		}

Asserts on the `client.Containers.ListContainersAsync` call

Python's docker package has no problem connecting to the docker machine.  My config is:

DOCKER_CERT_PATH=C:\Users\Marc\.docker\machine\machines\default
DOCKER_HOST=tcp://192.168.99.100:2376
DOCKER_MACHINE_NAME=default
DOCKER_TLS_VERIFY=1
DOCKER_TOOLBOX_INSTALL_PATH=C:\Program Files\Docker Toolbox

Any ideas?

**Comment by jterry75** [comment]
I am not 100% sure why you are seeing that. It doesn't actually repro for me. But a few things stand out. In your python client it looks like you are using certs to secure the tcp endpoint but in the `DockerClientConfiguration` you don't use any `X509Certificate2` credentials. Its possible that The code doesn't handle an access denied error well on the stream reader.

**Comment by galvesribeiro** [comment]
Yep, that error is because the Socket is being closed since the server (Docker daemon) is expecting the TLS handshake at this point.

As @jterry75  mentioned, please use the `X509Certificate2` credentials.

**Comment by MadhurBhardwaj** [comment]
I'm using C# lib (Docker.DotNet) to connect to Swarm manager , how to get the certificate file used for tls ?

**Comment by galvesribeiro** [comment]
@MadhurBhardwaj  you need to add an extra package to have the TLS credentials class.

Please have a look at this particular topic on our README page https://github.com/Microsoft/Docker.DotNet#example-https-authentication-to-docker

**Comment by MadhurBhardwaj** [comment]
Thanks @galvesribeiro , I was asking about how to find that certificate (file) used for tls, wher this file is stored , how to locate that ? any information on that ...

**Comment by galvesribeiro** [comment]
@MadhurBhardwaj you san use `File.ReadAllBytes()` than pass the `byte[]` to `X509Certificate2` constructors. There are multiple other ways like reading from the certificate store on the OS etc. Reading from the file is the easiest one.

**Comment by MadhurBhardwaj** [comment]
Thanks @galvesribeiro , question is how to get that file , I'm not asking about how to read....
where that file is stored , in certificate store ? user or localMachine ,, I'm accessing Swarm manager that is a remote machine from my local system , question is how to get this file so that I can access API's from my system using some application.....

**Comment by galvesribeiro** [comment]
@MadhurBhardwaj that is out of scope for this library actually. You can store it anywhere you want, and build the `X509Certificate2` class whatever way you prefer. It behave exactly the same way as if you where using the docker CLI.
